s3blobs/src/play/modules/s3blobs/S3Blob.java

```diff
@@ -116,16 +118,16 @@ public int hashCode(Object o) throws HibernateException {
 	}
 
 	@Override
-	public Object nullSafeGet(ResultSet rs, String[] names, Object o) throws HibernateException, SQLException {
-		String val = StringType.INSTANCE.nullSafeGet(rs, names[0]);
-		if (val == null || val.length() == 0 || !val.contains("|")) {
-			return new S3Blob();
+	public Object nullSafeGet(ResultSet rs, String[] names, SessionImplementor owner, Object o) throws HibernateException, SQLException {
+		String val = StringType.INSTANCE.nullSafeGet(rs, names[0], owner);
+		if (val == null || val.length() == 0 || !val.contains("|") || val.equals("null|null")) {
+			return null;
```

Comment:
Is this `nullSafeGet` not supposed to return `null`?  If not, maybe the right behavior is to throw an exception?